### PR TITLE
Type hash

### PR DESCRIPTION
**Pull request by tobyink**
This allows for hash-like flags (and args, but it seems more natural for flags).

    myscript.pl compile --define os=linux --define arch=x86_64 --define debug=1 input.bleh

It probably needs documenting too.

### CODE REVIEW

**Comment by tobyink** [comment]
This addresses issue #37.

**Comment by tobyink** [comment]
Yes. I will add more tests!

Are there any specific tests you think would be a good idea?

**Comment by tobyink** [comment]
Coverage for the patch is now 100%
https://codecov.io/gh/sago35/Getopt-Kingpin/pull/39

**Comment by sago35** [comment]
Please rebase

```
$ git checkout type-hash
$ git rebase master
```

**Comment by sago35** [comment]
This is a great improvement.
It's especially great that it's repeatable, as you wrote in t/29_hash.t.

I'd like to see some kind of information in the Getopt::Kingpin.pm pod.

**Comment by tobyink** [comment]
I can't figure out how to rebase this, but the branch doesn't conflict with master so should be able to be safely merged with master without a rebase.

File: lib/Getopt/Kingpin/Base.pm [hunk 1]
```diff
@@ -126,7 +131,20 @@ sub set_value {
         }
         push @values, $ret[0];
         $self->_defined(1);
-        $self->value([@values]);
+        $self->value(\@values);
```
Comment (sago35) [inline lib/Getopt/Kingpin/Base.pm:134]: Thanks for the improvement.

**Review by sago35 — APPROVED** [review]

**Comment by sago35** [comment]
Thank you for your contribution.
Great work.

**Pull request merged**